tests/unit/components/Navigator/QuickNavigationModal.spec.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2022 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { shallowMount } from '@vue/test-utils';
+import HighlightMatches from '@/components/Navigator/HighlightMatches.vue';
+import QuickNavigationModal from '@/components/Navigator/QuickNavigationModal.vue';
+
+describe('QuickNavigationModal', () => {
+  let wrapper;
+  let input;
+
+  const inputValue = 'foo';
+  const symbols = [
+    {
+      title: 'foo',
+      path: '',
+    },
+    {
+      title: 'fobaro',
+      path: '',
+    },
+    {
+      title: 'bar',
+      path: '',
+    },
+  ];
+  const filteredSymbols = [
+    symbols[0],
+    symbols[1],
+  ];
+  const config = {
+    propsData: {},
+    provide: {
+      quickNavigationStore: {
+        state: {
+          enableQuickNavigation: false,
+          flattenIndex: symbols,
+        },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    wrapper = shallowMount(QuickNavigationModal, config);
+    input = wrapper.find({ ref: 'input' });
+  });
+
+  it('it renders the Quick navigation modal', () => {
+    expect(input.exists()).toBe(true);
+    expect(wrapper.classes('quick-navigation-modal')).toBe(true);
+  });
+
+  it('it adds text to the input', async () => {
+    const textInput = wrapper.find('input');
+    await textInput.setValue(inputValue);
+    expect(wrapper.find('input').element.value).toBe(inputValue);
+  });
+
+  it('it filters the symbols on user input', async () => {
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    await wrapper.vm.$nextTick();
+    const matches = wrapper.vm.filteredSymbols;
+    expect(matches).toHaveLength(filteredSymbols.length);
```

Comment:
Consider adding a test checking that these symbols are actually being rendered.
You could check the number of `References` rendered and the url property passed through them.

Reply:
I added additional tests to check that the `Reference`component is being rendered with the corresponding symbol path in https://docs.google.com/spreadsheets/d/1izaWaP9a48_E4exGQtg3n35p18PXHLPy/edit?usp=sharing&ouid=101663954077174939249&rtpof=true&sd=true